src/tasks/gitea/gitea-drone-oauth.ts

```diff
@@ -0,0 +1,212 @@
+import * as k8s from '@kubernetes/client-node'
+import axios, { AxiosRequestConfig } from 'axios'
+import querystring from 'querystring'
+
+import { UserApi, CreateOAuth2ApplicationOptions, OAuth2Application } from '@redkubes/gitea-client-node'
+
+import { cleanEnv, GITEA_USER, GITEA_PASSWORD, GITEA_URL, GITEA_REPO } from '../../validators'
+
+const env = cleanEnv({
+  GITEA_USER,
+  GITEA_PASSWORD,
+  GITEA_URL,
+  GITEA_REPO,
+})
+type DroneSecret = {
+  clientId: string
+  clientSecret: string
+}
+class GiteaDroneOAuth {
+  apiClient: k8s.CoreV1Api
+  k8s_namespace = 'gitea'
+  k8s_secret_name = 'gitea-drone-secret'
+  csrfCookieName = '_csrf'
```

Comment:
please explain why _csrf cookie is required here

Reply:
The csrf cookie is used for the authorization/grant steps. It is required to be set accordingly for the "grant" to be approved.
I tested MVC (minimal viable cookies), and found that csrf was it.

Reply:
Great, could you add a brief comment in code, so in one month we know :)